examples/hdinsight/enterprise-security-package/main.tf

```diff
@@ -0,0 +1,257 @@
+    sync_ntlm_passwords     = true
+    sync_on_prem_passwords  = true
+    tls_v1_enabled          = true
+  }
+
+  tags = {
+    Environment = "test"
+  }
+
+  depends_on = [
+    azuread_group_member.example,
+    azurerm_role_assignment.example,
+    azurerm_subnet_network_security_group_association.example,
+  ]
+}
+
+resource "azurerm_virtual_network_dns_servers" "example" {
+  virtual_network_id = azurerm_virtual_network.example.id
+  dns_servers        = azurerm_active_directory_domain_service.example.initial_replica_set.0.domain_controller_ip_addresses
+}
+
+resource "azurerm_hdinsight_hadoop_cluster" "example" {
+  name                = "${var.prefix}-hdicluster"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  cluster_version     = "4.0"
+  tier                = "Premium"
+
+  component_version {
+    hadoop = "3.1"
+  }
+
+  gateway {
+    username = "sshuser"
+    password = "TerrAform123!"
+  }
+
+  storage_account {
+    storage_container_id = azurerm_storage_container.example.id
+    storage_account_key  = azurerm_storage_account.example.primary_access_key
+    is_default           = true
+  }
+
+  roles {
+    head_node {
+      vm_size            = "Standard_E4_V3"
+      username           = "sshuser"
+      password           = "TerrAform123!"
+      subnet_id          = azurerm_subnet.example.id
+      virtual_network_id = azurerm_virtual_network.example.id
+    }
+
+    worker_node {
+      vm_size               = "Standard_E8_V3"
+      username              = "sshuser"
+      password              = "TerrAform123!"
+      target_instance_count = 3
+      subnet_id             = azurerm_subnet.example.id
+      virtual_network_id    = azurerm_virtual_network.example.id
+    }
+
+    zookeeper_node {
+      vm_size            = "Standard_A2_V2"
+      username           = "sshuser"
+      password           = "TerrAform123!"
+      subnet_id          = azurerm_subnet.example.id
+      virtual_network_id = azurerm_virtual_network.example.id
+    }
+  }
+
+  security_profile {
+    aadds_resource_id       = "${replace(azurerm_active_directory_domain_service.example.id, "/initialReplicaSetId/${azurerm_active_directory_domain_service.example.deployment_id}", "")}"
+    domain_name             = azurerm_active_directory_domain_service.example.domain_name
+    domain_username         = azuread_user.example.user_principal_name
+    domain_user_password    = azuread_user.example.password
+    ldaps_urls              = ["ldaps://${azurerm_active_directory_domain_service.example.domain_name}:636"]
+    msi_resource_id         = azurerm_user_assigned_identity.example.id
+    cluster_users_group_dns = [azuread_group.example.display_name]
+  }
+
+  depends_on = [
+    azurerm_virtual_network_dns_servers.example,
+  ]
+
+  lifecycle {
+    ignore_changes = [
+      security_profile.0.domain_user_password,
+      gateway.0.password
+    ]
+  }
```

Comment:
We should remove this from the example too

Reply:
Updated